### PR TITLE
Fix profile watching if the profile file was never opened in the IDE

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Related Issue(s)
#1560

## Testing
Unit tests, manual testing using external text editor and alt tab back into IDE triggers refresh

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt [hunk 1]
```diff
@@ -4,35 +4,45 @@
...
     private var onUpdate: (() -> Unit)? = null
 
     init {
         Disposer.register(parentDisposable, this)
     }
 
     private val watchLocationsStrings = setOf(
-        FileUtil.normalize(ProfileFileLocation.configurationFilePath().toAbsolutePath().toString()),
-        FileUtil.normalize(ProfileFileLocation.credentialsFilePath().toAbsolutePath().toString())
+        ProfileFileLocation.configurationFilePath().toAbsolutePath().toString(),
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:30]: Why are we no longer calling normalize?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt [hunk 2]
```diff
@@ -4,35 +4,45 @@
...
+        LOG.debug { "Received events: $events" }
+        val isRelevant = events.any { watchLocationsStrings.contains(it.path) }
 
         return if (isRelevant) {
+            LOG.info { "Profile file change detected, scheduling refresh" }
             object : AsyncFileListener.ChangeApplier {
                 override fun afterVfsChange() {
-                    onUpdate?.invoke()
+                    // Off load this, since this is called under a write lock
+                    ApplicationManager.getApplication().executeOnPooledThread {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:43]: nit: could use coroutines

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt [hunk 3]
```diff
@@ -43,11 +53,33 @@ class ProfileWatcher(parentDisposable: Disposable) : AsyncFileListener, Disposab
...
+        LOG.info { "Starting profile watcher, profile locations: $watchLocationsStrings" }
+
+        val localFileSystem = LocalFileSystem.getInstance()
+
+        val watchLocationParents = watchLocationsStrings.map {
+            // Make VFS aware of it
+            localFileSystem.refreshAndFindFileByPath(it)
+
+            // Use the parent as the watch root in case file does not exist yet
+            // Note: This system requires that the parent folder already exists
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:65]: I don't think we should necessarily solve this, but what does JB do when it doesn't exist? create the file?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:65]: What do you mean? 

The find call returns null and the VFS ignores it so we won't receive any updates.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:30]: That was a mistake

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt (hunk 3 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:65]: Right that's what I mean, I think that that is fine, but I guess Intellij typically only calls this after it makes a file

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:66]: Ya

**Comment by abrooksv** [comment]
Going to move this over to feature/connection-ux and rebase on top of #1877 since these changes are breaking the ProfileCredentialProviderFactoryTest tests due to too strong of coupling

**Comment by abrooksv** [comment]
Updated PR to be based on feature/connection-ux

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt (hunk 1 above)
Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:37]: A mistake that we should never have called normalize?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt:37]: Mistake it was removed, it is back but in a different spot

**Comment by abrooksv** [comment]
Fixed profile watching on windows due to file normalization issues, changed to master

**Pull request merged**